docs/how-to-guides/data-basics/3-building-an-edit-form.md

```diff
@@ -0,0 +1,521 @@
+
+Here's how you can try it in your browser's dev tools:
+
+```js
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', 9 );  // Replace 9 with an actual page ID
+```
+
+Let's update `EditPageForm` accordingly:
+
+```js
+export function EditPageForm( { pageId, onCancel, onSaveFinished } ) {
+	const page = useSelect(
+		select => select( coreDataStore ).getEntityRecord( 'postType', 'page', pageId ),
+		[pageId]
+	);
+	return (
+		<div className="my-gutenberg-form">
+			<TextControl
+				label='Page title:'
+				value={ page.title }
+			/>
+			{ /* ... */ }
+		</div>
+	);
+}
+```
+
+Now it should look like that:
+
+![](./media/edit-form/form-populated.png)
+
+### Step 4: Making the Page title field editable
+
+There's one problem with our _Page title_ field: you can't edit it. It receives a fixed `value` but doesn't update it when typing. We need an `onChange` handler.
+
+
+You may have seen a pattern similar to this one in other React apps. It's known as a ["controlled component"](https://reactjs.org/docs/forms.html#controlled-components):
+
+```js
+export function VanillaReactForm({ initialTitle }) {
+	const [title, setTitle] = useState( initialTitle );
+	return (
+		<TextControl
+			value={ title }
+			onChange={ setTitle }
+		/>
+	);
+}
+```
+
+Updating entity records in Gutenberg Data is similar but instead of using `setTitle` to store in local state, we use the `editEntityRecord` action which stores the updates in the _Redux_ state. Here's how you can try it out in your browser's dev tools:
+
+```js
+// We need a valid page ID to call editEntityRecord, so let's get the first available one using getEntityRecords.
+const pageId = wp.data.select( 'core' ).getEntityRecords( 'postType', 'page' )[0].id;
+
+// Update the title
+wp.data.dispatch( 'core' ).editEntityRecord( 'postType', 'page', pageId, { title: 'updated title' } );
+```
+
+At this point, you may ask _how is `editEntityRecord` better than `useState`? The answer is that it offers a few features you wouldn't otherwise get.
+
+Firstly, we can save the changes as easily as we retrieve the data and ensure that all caches will be correctly updated.
+
+Secondly, the changes applied via `editEntityRecord` are easily undo-able via the `undo` and `redo` actions.
+
+Lastly, because the changes live in the _Redux_ state, they are "global" and can be accessed by other components. For example, we could make the `PagesList` display the currently edited title.
+
+To that last point, let's see what happens when we use `getEntityRecord` to access the entity record we just updated:
+
+```js
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', pageId ).title
+```
+
+It doesn't reflect the edits. What's going on?
+
+Well, let's imagine for a second that it would reflect the edits. If it did, then anything you typed in the title field would be _immediately_ displayed inside the `PagesList`.
```

Comment:
I think we need to add some wording here to clarify that the changes you type into the form field would be reflected in the PageList before you were ready to "commit" or persist them.

Reply:
Is it clearer now?